agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/ExampleExporter.java

```diff
@@ -0,0 +1,342 @@
+    private final SpanExporter delegate;
+
+    public static ExampleExporter create(SpanProcessorConfig config, SpanExporter delegate) {
+        // optimize data structure
+        optimizeSpanProcessorConfig(config);
+        return new ExampleExporter(config,delegate);
+    }
+
+    private static void optimizeSpanProcessorConfig(SpanProcessorConfig config) {
+        if(config!=null && config.actions!=null && config.actions.size()>0) {
+            config.insertActions = new ArrayList<>();
+            config.otherActions = new ArrayList<>();
+            for(SpanProcessorAction spanProcessorAction:config.actions) {
+                if(spanProcessorAction.action.equals("insert")) {
+                    config.insertActions.add(spanProcessorAction);
+                } else {
+                    config.otherActions.add(spanProcessorAction);
+                }
+            }
+        }
+    }
+
+    private ExampleExporter(SpanProcessorConfig config, SpanExporter delegate) {
+        this.config = config;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public CompletableResultCode export(Collection<SpanData> spans) {
+        // we need to filter attributes before passing on to delegate
+        List<SpanData> copy = new ArrayList<>();
+        for (SpanData span : spans) {
+            copy.add(process(span));
+        }
+        return delegate.export(copy);
+    }
+
+    private SpanData process(SpanData span) {
+
+        final String spanName = span.getName();
+        final ReadableAttributes existingSpanAttributes = span.getAttributes();
+        //Flag to check if the span is included in processing
+        Boolean includeFlag = null;
+        //Flag to check if the span is excluded in processing
+        Boolean excludeFlag = null;
+        if(config!=null) {
+            SpanData updatedSpan = processOtherActions(config,span,includeFlag,excludeFlag,spanName,existingSpanAttributes);
+            return processInsertActions(config,updatedSpan,includeFlag,excludeFlag,spanName,existingSpanAttributes);
+        }
+
+        return span;
+    }
+
+    private SpanData processInsertActions(SpanProcessorConfig config, SpanData span, Boolean includeFlag, Boolean excludeFlag, String spanName,
+                                          ReadableAttributes existingSpanAttributes) {
+
+        final Builder insertBuilder = Attributes.newBuilder();
+
+        if(config.insertActions!=null && config.insertActions.size()>0) {
+            if(includeFlag==null) {
+                includeFlag = checkIncludes(config.include,existingSpanAttributes,spanName);
+            }
+            if(excludeFlag==null) {
+                excludeFlag=checkExcludes(config.exclude,existingSpanAttributes,spanName);
+            }
+            if(includeFlag && !excludeFlag) {
+
+                for(SpanProcessorAction actionObj:config.insertActions) {
+                    if(actionObj.key!=null && actionObj.value!=null) {
```

Comment:
condition not needed if we remove insertActions from optimized model that are missing key or value